HpToolsLauncher/TestRunners/GuiTestRunner.cs

```diff
@@ -79,6 +79,31 @@ public TestRunResults RunTest(TestInfo testinf, ref string errorReason, RunCance
             ConsoleWriter.WriteLine(DateTime.Now.ToString(Launcher.DateFormat) + " Running: " + testPath);
             runDesc.ReportLocation = testPath;
 
+            // use the report path instead of the test path
+            // (if defined)
+            if (!String.IsNullOrEmpty(testinf.ReportPath))
+            {
+                runDesc.ReportLocation = testinf.ReportPath;
+
+                if (!Directory.Exists(runDesc.ReportLocation))
+                {
+                    try
+                    {
+                        Directory.CreateDirectory(runDesc.ReportLocation);
+                    }
+                    catch (Exception)
+                    {
+                        errorReason = string.Format(Resources.InvalidReportPath, runDesc.ReportLocation);
+                        runDesc.TestState = TestState.Error;
+                        runDesc.ErrorDesc = errorReason;
+                        runDesc.TestPath = testinf.TestPath;
+                        ConsoleWriter.WriteErrLine(runDesc.ErrorDesc);
+                        ConsoleWriter.ErrorSummaryLines.Add(runDesc.ErrorDesc);
+                        Environment.ExitCode = (int)Launcher.ExitCodeEnum.Failed;
+                        return runDesc;
+                    }
+                }
+            }
```

Comment:
Just confirming is the else clause not needed in this file comparing to ApiTestRunner.cs file?

Reply:
Now no else is required as the default report path is set before trying to set the custom report path.